### PR TITLE
Panel: support drag-resizing

### PR DESCRIPTION
**Pull request by kevingraber**
# Details
### Description
Adds the ability to resize the `Panel` component.

### Comments
- Currently experiencing an issue resizing the panel when attach is set to right. It seems to expand / shrink from the right side only.
- Should the we allow the panel to resize itself in an uncontrolled manor?
- Also currently trying to get the right side content to resize at the same speed as the panel.

### Meta

**_Stakeholders:_** @Vimeo/design-systems-core

# Screenshots
Remember to attach screenshots here!

### CODE REVIEW

**Review by seanmcintyre — CHANGES_REQUESTED** [review]
More questions/thoughts: 

What is the reasoning behind managing the width externally? A width determined by dragging seems like it would be necessarily always intrinsic.

I think you could simplify the dragging management and cease to rely on binding event listeners by combining `draggable`, `onDragStart`, `onDragEnd`, and `onDrag` on the `<DragEdge />` component if you make it invisible. The `onDrag` event should provide cursor coordinates that you can use to handle the resizing.

Don't forget dark mode support!

I think the animation could use some work. It's kind of janky, some elements flicker, and some animate at very different speeds. This is easier to see if you slow it down in devtools:

https://user-images.githubusercontent.com/319302/129091185-96ef7c12-de1f-4087-915e-192fbf9b90e9.mov

File: src/layout/Panel/Panel.examples.story.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        }}
+        onDragBegin={() => {
+          setIsDragging(true);
+        }}
+        onDragEnd={() => setIsDragging(false)}
+        resizable={true}
+        screen={false}
+        style={{ width: navWidth }}
+      >
+        {/* TODO - Panel needs a child - is it ok to use a fragment? */}
```
[the remark this answers is not in the document]
Comment (kevingraber) [reply src/layout/Panel/Panel.examples.story.tsx:74]: I was, but after visiting the `Panel.types.ts` file it went away.

**Comment by kevingraber** [comment]
I tried using the `draggable` attribute but wasn't able to get a good UX. The dragging was a lot choppier, there were still remnants of the dragging UI (cursor icon), and the drag edge would still be active for a second or two after the drag ended.

**Comment by kevingraber** [comment]
Thanks for all the feedback @seanmcintyre. I started looking into the panel animation and found the following.

https://user-images.githubusercontent.com/15743757/129586470-10f33168-0db3-4ab4-931c-cd668b37d29d.mov

It looks like if I close the panel from outside, the animation works as expected. If I close it using the button from the inside, the entire panel disappears before it is entirely off the screen. Also if I set focus on an element inside the panel and then close it using the external button, the panel exhibits the same behavior of closing before it's fully off the screen. I don't think this has anything to with any changes I made. Any thoughts on what could be happening here?

Also I believe my builds are failing because of lint issues in files I haven't touched?

**Comment by kevingraber** [comment]
@seanmcintyre everything looks great to me! Design signed off as well.

**Review by seanmcintyre — APPROVED** [review]

**Pull request merged**